core/src/main/scala/org/apache/spark/sql/SnappyParser.scala

```diff
@@ -103,10 +103,18 @@ class SnappyParser(session: SnappySession)
     var index = 0
     val len = s.length
     // use double if ending with D/d, float for F/f and long for L/l
+
     s.charAt(len - 1) match {
       case 'D' | 'd' =>
-        return newTokenizedLiteral(
-          java.lang.Double.parseDouble(s.substring(0, len - 1)), DoubleType)
+        if (s.length > 2 && (s.charAt(len - 2) match {
+          case 'B' | 'b' => true
+          case _ => false
+        })) {
+          return toDecimalLiteral(s.substring(0, len - 2), checkExactNumeric = false)
+        } else {
+          return newTokenizedLiteral(
+            java.lang.Double.parseDouble(s.substring(0, len - 1)), DoubleType)
+        }
```

Comment:
Add checks for valid suffixes too:

```scala
if (s.length > 2) {
  s.charAt(len - 2) match {
    case 'B' | 'b' => toDecimalLiteral(...)
    case c if Character.isDigit(c) => newTokenalizedLiteral(...)
    case _ => throw new ParseException(...)
  }
}
```

Likewise for non 'D' | 'd' cases, the previous char should be checked that it should be a digit and never a numericSuffix -- see next comment.